chocolatey-community/chocolatey-community-validation
# (#28) Add rule to disallow icon URL using GitHub links

## Description Of Changes

This introduces a new requirement rule that flags any icon URL that
makes use of GitHub, either usercontent, gists or the main domain as
well as the rawgit domain that is no longer possible to use properly.

## Motivation and Context

We want a rule that flags and prevents using GitHub raw urls.

## Testing

1. Create a new nuspec using `choco new`.
2. Update the Icon URL to be `https://github.com/chocolatey-community/chocolatey-packages/blob/master/icons/7zip.svg`
3. Run `choco pack` against the created nuspec.
4. Verify the rule ID `CPMR0076` is flagged and shown in the console.
5. Repeat 2-4 using the URL `https://github.com/chocolatey-community/chocolatey-packages/raw/refs/heads/master/icons/filezilla.svg`
6. Repeat 2-4 using the RUL `https://raw.githubusercontent.com/chocolatey-community/chocolatey-packages/refs/heads/master/icons/1password4.png`
7. Repeat 2-4 using the URL `https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/049a3a3d/icons/winff.png`

### Operating Systems Testing

## Change Types Made


## Change Checklist


## Related Issue

Fixes #28
